### PR TITLE
Add term entry kotlin datetime method todayin 3866

### LINKED ISSUE #3866: [Term Entry] Kotlin Datetime .todayIn()
### Reason/inspiration (optional)

We would like a new term entry on the `.todayIn()` method for Kotlin: [Datetime](https://www.codecademy.com/resources/docs/kotlin/datetime). The entry should go in a new file under `docs/content/kotlin/concepts/datetime/terms/todayIn/todayIn.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```kotlin `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by aa-arroyo**
### Type of Change

- Adding a new entry: I am adding a term entry (#3866). Kotlin datetime, todayIn() method. I have created a directory named todayIn and a todayIn.md file. I am working on this entry as part of the Learn Git & GitHub course at Codecademy. 
This is my first time learning about Kotlin/Kotlinx. Setting up the IntelliJ IDEA IDE and adding the kotlinx.datetime dependency took me some time, but I eventually got it to work and could play around with the code and better understand it. Any guidance and feedback will be much appreciated.

Closes #3866 

### Checklist

- [Y] All writings are my own.
- [Y] My entry follows the Codecademy Docs style guide.
- [Y] My changes generate no new warnings.
- [Y] I have performed a self-review of my own writing and code.
- [Y] I have checked my entry and corrected any misspellings.
- [Y] I have made corresponding changes to the documentation if needed.
- [Y] I have confirmed my changes are not being pushed from my forked `main` branch.
- [Y] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [Y] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/todayIn/todayIn.md:14]: Remove the comments from the metadata as those comments are only for your convenience.

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/todayIn/todayIn.md:20]:
```suggestion
```
Remove the extra space between the lines.

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/todayIn/todayIn.md:23]: Please go through the [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md) and format the code blocks and syntax as mentioned.

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/todayIn/todayIn.md:28]:
```suggestion
```

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/todayIn/todayIn.md:38]: The same goes for this code, go through the style guide and format this code block correctly. Display the output in a different code block, you can use the shell for it.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@aa-arroyo First of all, my suggestion would be that go through other entries of the docs, and observe how the content and code blocks are formatted. Take reference from the style guide and make the changes in this file.

**Comment by aa-arroyo** [comment]
avdhoottt, thank you for the detailed feedback. I will work on all the requested changes.

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (aa-arroyo) [reply content/kotlin/concepts/datetime/terms/todayIn/todayIn.md]: Fixed.

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (aa-arroyo) [reply content/kotlin/concepts/datetime/terms/todayIn/todayIn.md]: Fixed.

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (aa-arroyo) [reply content/kotlin/concepts/datetime/terms/todayIn/todayIn.md]: I reviewed the style guide and made the necessary corrections to the todayIn.md file.

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (aa-arroyo) [reply content/kotlin/concepts/datetime/terms/todayIn/todayIn.md]: Fixed.

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (aa-arroyo) [reply content/kotlin/concepts/datetime/terms/todayIn/todayIn.md]: Fixed.

**Comment by aa-arroyo** [comment]
@avdhoottt I have corrected the todayIn.md file according to the feedback provided and the style guide. I added, committed, and pushed my changes. Please let me know if any additional actions are required from me, and I will get it done as soon as possible.

Thank you.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@aa-arroyo I appreciate your work but I said to remove the line spaces from the code only and keep the other required spaces. For example, add a line space before and after the syntax and example code blocks. Please add the required spaces and it's good to go for the second review.

**Comment by aa-arroyo** [comment]
@avdhoottt Added the required spacing.

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/todayIn/todayIn.md:38]:
```suggestion
import kotlinx.datetime.* 

fun main() {
  val myLocalDate = Clock.System.todayIn(timeZone = TimeZone.currentSystemDefault())
  println(myLocalDate)
}
```

Comment (aa-arroyo) [reply content/kotlin/concepts/datetime/terms/todayIn/todayIn.md]: Removed blank line 41.

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/todayIn/todayIn.md:18]:
```suggestion
## Syntax
```

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/todayIn/todayIn.md:26]:
```suggestion
```

**Review by avdhoottt — APPROVED** [review]
Looks Good for the second review.

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (SSwiniarski) [inline content/kotlin/concepts/datetime/terms/todayIn/todayIn.md:3]:
```suggestion
Description: 'Returns a date when calling this method on a Clock with a specific time zone as an argument.'
```

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (SSwiniarski) [inline content/kotlin/concepts/datetime/terms/todayIn/todayIn.md:16]:
```suggestion
The **`.todayIn()`** method returns a date when called on a `Clock` while passing a `TimeZone` argument.
```
A clock isn't an instant value, it is a source of instant values.

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (SSwiniarski) [inline content/kotlin/concepts/datetime/terms/todayIn/todayIn.md:21]:
```suggestion
myClock.todayIn(myTimeZone)
```
The Syntax example needs to be more generic.  (It can be any Clock and any TimeZone)

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (SSwiniarski) [inline content/kotlin/concepts/datetime/terms/todayIn/todayIn.md:24]:
```suggestion
- `myClock`: is a Clock providing a source for Instants (e.g., 2023-11-21, 00:00:00 UTC+0). `Clock.System` is the Clock that refers to the current platform for the current time.
```

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (SSwiniarski) [inline content/kotlin/concepts/datetime/terms/todayIn/todayIn.md:25]:
```suggestion
- `myTimeZone`: An instance of the TimeZone class. (e.g., America/Los_Angeles).
```

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (SSwiniarski) [inline content/kotlin/concepts/datetime/terms/todayIn/todayIn.md:28]: There should be an introductory sentence summarizing what the example is doing.

File: content/kotlin/concepts/datetime/terms/todayIn/todayIn.md
Comment (SSwiniarski) [inline content/kotlin/concepts/datetime/terms/todayIn/todayIn.md:39]:
```suggestion
This example results in something like the following output:
```
The code as written will not always return the same date.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@aa-arroyo I did a second review and had a number of changes to suggest. Especially to clean up the confusion between a Clock and an Instant, they're two different things.

**Comment by aa-arroyo** [comment]
Hi @SSwiniarski 

I made all the suggested changes and added an introductory sentence summarizing the example. I tried pushing my changes like I did before, but I got this error:

To https://github.com/aa-arroyo/docs.git
 ! [rejected]          add-term-entry-kotlin-datetime-method-todayin-3866 -> add-term-entry-kotlin-datetime-method-todayin-3866 (non-fast-forward)
error: failed to push some refs to 'https://github.com/aa-arroyo/docs.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. If you want to integrate the remote changes,
hint: use 'git pull' before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

I tried `git pull origin main` and then tried pushing from the feature branch (as I have been doing until now), but it's still not working.

On my forked feature branch under the Sync Fork option, it reads as follows: This branch is not behind the upstream Codecademy:main
No new commits to fetch. Enjoy your day!

I followed the steps listed on the link below, but it did not fix the error.
https://docs.github.com/en/get-started/using-git/dealing-with-non-fast-forward-errors

I am a bit lost. Any help will be appreciated.

**Comment by SSwiniarski** [comment]
@caupolicandiaz , @KTom101 any idea on how to resolve their problem?

**Comment by caupolicandiaz** [comment]
> Hi @SSwiniarski
> 
> I made all the suggested changes and added an introductory sentence summarizing the example. I tried pushing my changes like I did before, but I got this error:
> 
> To https://github.com/aa-arroyo/docs.git ! [rejected] add-term-entry-kotlin-datetime-method-todayin-3866 -> add-term-entry-kotlin-datetime-method-todayin-3866 (non-fast-forward) error: failed to push some refs to 'https://github.com/aa-arroyo/docs.git' hint: Updates were rejected because the tip of your current branch is behind hint: its remote counterpart. If you want to integrate the remote changes, hint: use 'git pull' before pushing again. hint: See the 'Note about fast-forwards' in 'git push --help' for details.
> 
> I tried `git pull origin main` and then tried pushing from the feature branch (as I have been doing until now), but it's still not working.
> 
> On my forked feature branch under the Sync Fork option, it reads as follows: This branch is not behind the upstream Codecademy:main No new commits to fetch. Enjoy your day!
> 
> I followed the steps listed on the link below, but it did not fix the error. https://docs.github.com/en/get-started/using-git/dealing-with-non-fast-forward-errors
> 
> I am a bit lost. Any help will be appreciated.

I believe you want to do `git pull origin feature-branch-name`, not main. Then, you can commit the changes pulled down. After the changes are downloaded the window will switch to the vim editor, use character `I` to type your commit message. Hit escape and then `:wq` to exit. Hopefully, that resolves your situation.

**Comment by aa-arroyo** [comment]
Hi @caupolicandiaz 

I tried the suggested command and followed the hints/instructions provided by git:

mypc todayIn % git pull origin add-term-entry-kotlin-datetime-method-todayin-3866         
From https://github.com/aa-arroyo/docs
 * branch              add-term-entry-kotlin-datetime-method-todayin-3866 -> FETCH_HEAD
hint: You have divergent branches and need to specify how to reconcile them.
hint: You can do so by running one of the following commands sometime before
hint: your next pull:
hint: 
hint:   git config pull.rebase false  # merge
hint:   git config pull.rebase true   # rebase
hint:   git config pull.ff only       # fast-forward only
hint: 
hint: You can replace "git config" with "git config --global" to set a default
hint: preference for all repositories. You can also pass --rebase, --no-rebase,
hint: or --ff-only on the command line to override the configured default per
hint: invocation.
fatal: Need to specify how to reconcile divergent branches.


mypc todayIn % git pull --ff-only origin add-term-entry-kotlin-datetime-method-todayin-3866
From https://github.com/aa-arroyo/docs
 * branch              add-term-entry-kotlin-datetime-method-todayin-3866 -> FETCH_HEAD
hint: Diverging branches can't be fast-forwarded, you need to either:
hint: 
hint: 	git merge --no-ff
hint: 
hint: or:
hint: 
hint: 	git rebase
hint: 
hint: Disable this message with "git config advice.diverging false"
fatal: Not possible to fast-forward, aborting.


mypc todayIn % git merge --no-ff origin add-term-entry-kotlin-datetime-method-todayin-3866
Already up to date.

mypc todayIn % git status
On branch add-term-entry-kotlin-datetime-method-todayin-3866
nothing to commit, working tree clean

There was no Vim window. I tried pushing again and got the same error as before.

**Comment by aa-arroyo** [comment]
feature-branch-history

I got even more confused looking at this graph on VScode... It seems there are two feature branches with the same name (?). It seems I messed up somewhere along the way, I just don't know exactly how.

**Comment by caupolicandiaz** [comment]
@SSwiniarski I went ahead and pushed the changes. 
@aa-arroyo Unfortunately, it can be very difficult to troubleshoot an issue remotely. I would advise that you review our documentation for making and updating a PR. And try again on another entry.

**Review by SSwiniarski — APPROVED** [review]
@aa-arroyo Looks good.

**Comment by aa-arroyo** [comment]
@SSwiniarski @caupolicandiaz I understand. Thank you for all the help along the way! Sorry, I couldn't produce the best entry version on the first attempt. 

Cheers.

**Pull request merged**
